### PR TITLE
Switch to Python 3.8

### PR DESCRIPTION
**Pull request by diox**
Fixes https://github.com/mozilla/addons-server/issues/11980

### CODE REVIEW

**Comment by diox** [comment]
@eviljeff and @willdurand you might also be interested in having a look at this. It touches bad zip file handling in validation task decorator, but that seems ok.

File: src/olympia/devhub/tasks.py [hunk 1]
```diff
@@ -180,11 +180,12 @@ def wrapper(task, results, pk, *args, **kwargs):
...
-        except BadZipfile:
+        except BadZipFile:
+            # If we raised a BadZipFile we can return a single exception with
+            # a generic message indicating the zip is invalid or corrupt.
             results = deepcopy(amo.VALIDATOR_SKELETON_EXCEPTION_WEBEXT)
-            annotations.insert_validation_message(
-                results, type_='error',
-                message=ugettext('Invalid or corrupt add-on file.'))
+            results['messages'][0]['message'] = ugettext(
+                'Invalid or corrupt add-on file.')
```
Comment (eviljeff) [inline src/olympia/devhub/tasks.py:188]: Why can't we use annotations here any longer? (it's still used a few lines above)

**Comment by diox** [comment]
> it's mainly just changing `BadZipfile` to `BadZipFile`?

I didn't have to do that, in fact. Both still work, but while I was looking at the documentation I noticed it was deprecated so I figured I might as well take care of it now.

File: src/olympia/devhub/tasks.py (hunk 1 above)
Comment (diox) [reply src/olympia/devhub/tasks.py:188]: Because I don't want to insert a new error, I want to replace the base one. 

The main difference with before is that before, we ended up catching some invalid zip files later, when validating the files inside, and we raised a `ValidationError`. That was turned by this decorator into a generic linter error using the code just below this:
```python
        except Exception as exc:
            log.exception('Unhandled error during validation: %r' % exc)
            results = deepcopy(amo.VALIDATOR_SKELETON_EXCEPTION_WEBEXT)
            return results
```

Note that in that case, there is only one error returned. 

Now, with Python 3.8, `BadZipFile` is raised earlier than before for the same invalid zips. So we end up in that `except` clause, where we previously returned *two* errors: `Invalid or corrupt add-on file` *and* the generic ` coming from `VALIDATOR_SKELETON_EXCEPTION_WEBEXT`, `Sorry, we couldn't load your WebExtension.`. To stay consistent with what we were doing before for those zips, I'm changing the way we deal with `BadZipFile` and only return *one* error in that case, the `Invalid or corrupt add-on file`. This means I can no longer use `insert_validation_message`, as that would add a new error message instead of replacing the generic one.

File: src/olympia/files/utils.py [hunk 2]
```diff
@@ -742,16 +742,12 @@ def __init__(self, source, mode='r', force_fsync=False):
...
         self.force_fsync = force_fsync
-        self.is_valid = self.initialize_and_validate()
+        self.initialize_and_validate()
 
     def initialize_and_validate(self):
         """
         Runs some overall archive checks.
         """
-        # Shortcut to avoid expensive check over and over again
-        if getattr(self, 'is_valid', False):
```
Comment (bobsilverberg) [inline src/olympia/files/utils.py:752]: Why did you remove this optimization? Did it never get used?

Comment (diox) [reply src/olympia/files/utils.py:752]: It wasn't - if you look at the whole patch, there is only one place where we used `is_valid` left (that I removed), and even there it was dead code, because an exception would have be thrown before we got to that point.

**Comment by diox** [comment]
I'll merge this after the tag just to be safe.

File: src/olympia/files/utils.py (hunk 2 above)
Comment (bobsilverberg) [reply src/olympia/files/utils.py:752]: Right, but wasn't the idea of this that if we had to call `initialize_and_validate` on the same object more than once we could shortcut it? I assume there's nowhere in the code where `initialize_and_validate` might be called more than once on the same object?

Comment (diox) [reply src/olympia/files/utils.py:752]: It's called at `__init__` time so there is no way around it (I think historically it wasn't always the case, and that's why this property made sense)
